app/controllers/storage_locations_controller.rb

```diff
@@ -11,7 +11,14 @@ def quantity
   def index
     @selected_item_category = filter_params[:containing]
     @items = current_organization.storage_locations.items_inventoried
-    @storage_locations = current_organization.storage_locations.alphabetized.includes(:inventory_items).class_filter(filter_params)
+
+    @include_inactive_storage_locations = params[:include_inactive_storage_locations].to_i > 0
+
+    @storage_locations = if @include_inactive_storage_locations
```

Comment:
To keep this a bit more "DRY", I think you could define @storage_location and then conditionally add chain `all` or `kept`. This is a bit more clearer IMO since I understand that this conditional is really about which to chain on.

Reply:
Are you suggesting something like this:

```ruby
@include_inactive_storage_locations = params[:include_inactive_storage_locations].to_i
@storage_locations = current_organization.storage_locations.alphabetized.includes(:inventory_items).class_filter(filter_params)

@storage_locations = @storage_locations.kept if @include_active_storage_locations > 0 # potentially another query
```

Maybe I am incorrect, but wouldn't that cause us to make an unnecessary query?

Is the following any clearer? I don't think it would result in an extra query.

```ruby
@include_inactive_storage_locations = params[:include_inactive_storage_locations].to_i
@storage_locations = @include_inactive_storage_locations > 0 ? 
  current_organization.storage_locations.alphabetized.includes(:inventory_items).class_filter(filter_params) :
  current_organization.storage_locations.alphabetized.includes(:inventory_items).class_filter(filter_params).kept
```

Reply:
I'am not entirely sure if `class_filter` actually does the query but adding `.kept` prior to the class filters shouldn't cause extra queries. It usually is smart enough not to make a query until you are reading the results of the query. Could you double-check that?

Here is a slightly different way:
```
@storage_locations = current_organization.storage_locations.alphabetized
if @include_active_storage_locations > 0
   @storage_location = @storage_location.kept
else
   @storage_location = @storage_location.all
end

@storage_location.class_filter(filter_params) # put this at the end in case it causes the query to trigger
```

This reduces a bit of duplication and allows the block or conditional to be very clear about what it is doing. Let me know what you think.